### PR TITLE
Refactor umfPoolCreate to accept only one provider

### PR DESCRIPTION
**Pull request by vinser52**
Pool is a combination of memory provider and pool manager. The umfPoolCreate function s is refactored to accept only a single memory provider as we agreed.

### CODE REVIEW

**Comment by vinser52** [comment]
Should we update the UR repo as well?

**Comment by igchor** [comment]
> Should we update the UR repo as well?

For UR, we actually use some C++ wrappers so it might be enough to just update those - I'll take a look at this once DisjointPool is merged.

File: include/umf/memory_pool.h [hunk 1]
```diff
@@ -137,15 +135,12 @@ umf_memory_pool_handle_t umfPoolByPtr(const void *ptr);
...
-/// \param numProviders [in] number of memory providers to return
-/// \param numProvidersRet pointer to the actual number of memory providers
+/// \param hProvider [out] memory providers handle.
 /// \return UMF_RESULT_SUCCESS on success or appropriate error code on failure.
+///         UMF_RESULT_ERROR_INVALID_ARGUMENT if hProvider is NULL
 enum umf_result_t
-umfPoolGetMemoryProviders(umf_memory_pool_handle_t hPool, size_t numProviders,
-                          umf_memory_provider_handle_t *hProviders,
-                          size_t *numProvidersRet);
+umfPoolGetMemoryProviders(umf_memory_pool_handle_t hPool,
```
Comment (igchor) [inline include/umf/memory_pool.h:142]: Should we change the name to `umfPoolGetMemoryProviders`? (without the `s`)

**Comment by vinser52** [comment]
> > Should we update the UR repo as well?
> 
> For UR, we actually use some C++ wrappers so it might be enough to just update those - I'll take a look at this once DisjointPool is merged.

If disjointPool merged before this PR, I will update this PR to fix DisjointPool if needed.

**Comment by igchor** [comment]
Btw, @ldorau from which UR commit have you taken the umf code? E.g. I don't see the memory tracker (as in this version: https://github.com/oneapi-src/unified-runtime/tree/88c7f8f4e3db8a22880140ea14098c6572c4e2de/source/common/unified_malloc_framework/src) - will this be added later?

File: include/umf/memory_pool.h (hunk 1 above)
Comment (vinser52) [reply include/umf/memory_pool.h]: Done

**Review by igchor — APPROVED** [review]

**Review by bratpiorka — APPROVED** [review]

**Pull request merged**